packages/transformers/posthtml/src/PostHTMLTransformer.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+import {Transformer} from '@parcel/plugin';
+
+import posthtml from 'posthtml';
+import parse from 'posthtml-parser';
+import render from 'posthtml-render';
+import nullthrows from 'nullthrows';
+import semver from 'semver';
+import loadPlugins from './loadPlugins';
+
+export default new Transformer({
+  async getConfig({asset}) {
+    let config = await asset.getConfig(
+      ['.posthtmlrc', '.posthtmlrc.js', 'posthtml.config.js'],
+      {
+        packageKey: 'posthtml'
+      }
+    );
+
+    config = config || {};
+
+    // load plugins
+    config.plugins = await loadPlugins(config.plugins, asset.filePath);
+
+    // tells posthtml that we have already called parse
+    config.skipParse = true;
+    return config;
+  },
+
+  canReuseAST({ast}) {
+    return ast.type === 'posthtml' && semver.satisfies(ast.version, '^0.4.0');
+  },
+
+  async parse({asset, config}) {
+    // if we don't have a config it is posthtml is not configure, don't parse
+    if (!config) {
+      return;
+    }
+
+    return {
+      type: 'posthtml',
+      version: '0.4.1',
+      program: parse(await asset.getCode(), {
+        lowerCaseAttributeNames: true
+      })
+    };
+  },
+
+  async transform({asset, config}) {
+    if (!config) {
+      return [asset];
+    }
+
+    const ast = nullthrows(asset.ast);
+
+    let res = await posthtml(config.plugins).process(ast.program, config);
+
+    if (res.messages) {
+      await Promise.all(
```

Comment:
doesn't look like this needs to be async right?

Reply:
`asset.addConnectedFile` returns a promise, so I thought they needed to be awaited.

Reply:
oh you're right. I forgot about that.